sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java

```diff
@@ -478,7 +478,7 @@ public void testInternalConnect_setDirectRunner() throws Exception {
 
   @Test
   public void testInternalConnect_setBogusRunner() throws Exception {
-    thrown.expectMessage("Unknown 'runner' specified 'bogus'");
+    thrown.expectMessage("Error while executing SQL \"SELECT * FROM test\"");
```

Comment:
This looks to be a regression in Avatica which is dropping exception messages. I would guess it blocks this upgrade. I've filed a bug: https://issues.apache.org/jira/browse/CALCITE-3040

Reply:
[It ](https://github.com/apache/calcite-avatica/pull/84/files#diff-790bbb8c94560ecaff0907e1008fb01e)does drop some exception stack messages. thanks for filing the bug!

Reply:
It seems like Avatica hid e.getMessage() in SQLException message.
For us, it should not only assert error message, but also assert exception / throwable.

```java
thrown.expectMessage("Error while executing SQL \"SELECT * FROM test\"");
thrown.expectCause(IsInstanceOf.<Throwable>instanceOf(IllegalArgumentException.class));
```

Reply:
This test is checking to see if the error message gets propagated through in the message string, which is all most JDBC clients display. It is testing exactly what it should be. This will be fixed in Avatica 1.15.0, which is coming soon.